store/store.go

```diff
@@ -393,12 +393,26 @@ func (bs *BlockStore) SaveBlock(block *types.Block, blockParts *types.PartSet, s
 	if block == nil {
 		panic("BlockStore can only save a non-nil block")
 	}
-	if err := bs.saveBlockToBatch(block, blockParts, seenCommit); err != nil {
+
+	batch := bs.db.NewBatch()
+	defer func(batch dbm.Batch) {
+		err := batch.Close()
+		if err != nil {
+			panic(err)
+		}
+	}(batch)
```

Comment:
I would extract this as a function:

```golang
func closeBatchOrPanic(batch dbm.Batch) {
	if err := batch.Close(); err != nil {
		panic(err)
	}
}
```

Then you could simply:

```suggestion
	defer closeBatchOrPanic(batch)
```

Makes the enclosing function a little easier to read. Similarly for this same type of pattern in other packages.